### ISSUE TITLE
feature: use trichotomy rather than dichotomy for downsynced object

### ISSUE THREAD

**Issue by MikeSpreitzer** [kubestellar/kubestellar#1112]
### Feature Description

Currently a downsynced object is divided into two parts:
- desired state, which travels from core to WEC, and
- reported state, which travels from WEC to core.

Actually, there is a third part: system state, which is specific to the space and controlled by the apiserver and does not travel (e.g., UID, ResourceVersion).

The feature here is to introduce a third non-system part, state that does not travel. I have heard of two motivations so far.
1. Nick Mitchell wants to have some derived state that is large, derived by a controller in the core without a spurious copy to the WEC.
2. @clubanderson wants to have an object introduced in the core but a part of the object created in the WEC and not transported back to the core.

### Proposed Solution

Introduce a third non-system part of downsynced objects and an API by which users can identify it.

### Want to contribute?

- [ ] I would like to work on this issue.

**Comment by clubanderson** [comment]
@MikeSpreitzer the point of #1094 is to turn off the overwrite of an object after it is initially placed by KubeStellar.  I still want status to return for that object.  The point of the issue I opened is that beyond the initial placement of the object, I do not want KubeStellar to overwrite it.  This comes in handy when I deploy a secret without sensitive information (as a placeholder) for the operator of the WEC to fill in.  That sensitive information should not be visible in our MB or WDS and we should not overwrite the object with KS syncer.  It might be useful to have a way to forcefully overwrite the object in certain situations, but that is not in scope for today.

**Comment by MikeSpreitzer** [comment]
@clubanderson : yes, that is covered by what this Issue is talking about. This issue also covers what Nick is talking about.

**Comment by MikeSpreitzer** [comment]
Also, this issue is close to an aspiration that I have not tried to address yet. You can see it in the "One-to-many" section of https://medium.com/@mspreitz/seven-ways-to-stub-your-toes-on-the-edge-c45e060bc4f4 , where I talk about the need for imprecision. In some cases the trichotomy idea in this Issue covers the need for imprecision; those are the cases when KubeStellar can create the object in the WEC with the imprecise parts absent. OTOH, if an imprecise part has to be created with an initial or default value, that is not the same as marking that part as immobile.
